### PR TITLE
27 board class refactor update board method to use player type

### PR DESCRIPTION
**Pull request by cariperi**
This PR will refactor the 'update board human' and 'update board computer' methods in the Board class to be one single method, 'update board'. The new method takes two arguments: the player's selection, and the current player's type (human or computer). 

We updated the tests to reflect this refactor. Additionally, we are adding the current_player attribute as an instance variable for the ConnectFour class, and adding a method to ConnectFour that sets and changes the current player from human to computer.  

This additional functionality will also be used to refactor and DRY up the play method in the Connect Four class (see Issue #26).

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -27,12 +27,9 @@ def format_board
...
   end
 
-  def update_board_human(player_selection)
-    @grid[player_selection][@grid[player_selection].rindex('.')] = 'X'
-  end
-
-  def update_board_computer(player_selection)
-    @grid[player_selection][@grid[player_selection].rindex('.')] = 'O'
+  def update_board(player_selection, player_type)
+    player_type == :human ? piece = 'X' : piece = 'O'
```
Comment (KatieLonsdale) [inline lib/board.rb:31]: can we pass current player in this argument and use player.type or player.piece?

File: lib/connect_four.rb [hunk 2]
```diff
@@ -63,13 +70,13 @@ def play
     puts @board.format_board
 
     until @game.game_over == true
-      @board.update_board_human(@game.get_human_selection)
+      @board.update_board(@game.get_human_selection, :human)
```
Comment (KatieLonsdale) [inline lib/connect_four.rb:73]: Can we pass current player here?

**Review by KatieLonsdale — CHANGES_REQUESTED** [review]

File: lib/board.rb (hunk 1 above)
Comment (cariperi) [reply lib/board.rb]: done! I've updated the method to take the current_player as an argument, and then call current_player.piece inside the method to get the X or O.

File: lib/connect_four.rb (hunk 2 above)
Comment (cariperi) [reply lib/connect_four.rb]: done! I had to add a few extra lines in the play method to switch the current player back and forth -- this is hopefully temporary, and will be refactored out in the next issue (#26).

**Comment by cariperi** [comment]
I made the changes to pass the current_player value in to the update_board method, and use the Player class' .piece method. Thanks for the suggestions!

I also checked the Board class tests against existing methods, and added edge case tests for no-win and no-loss conditions (we'd had a comment to come back and do so!).  See [the last commit here](https://github.com/KatieLonsdale/connect_four/pull/28/commits/744fe5f7f2d82bc6cf8b9d38c7fde829811e24f5) .

**Review by KatieLonsdale — APPROVED** [review]
